### PR TITLE
anki-bin: set correct name and version for derivation

### PR DESCRIPTION
**Pull request by cdepillabout**
###### Description of changes

`anki-bin` is built with `buildFHSUserEnv` on Linux, which doesn't set a `version` for the resulting derivation.  This commit overrides the resulting derivation to have a version set correctly.

This is important for end-users to be able to easily tell what version of Anki they will get when they install `anki-bin`.  It is normally very important for an end-user to use the correct version of Anki.

`anki-bin` not having a version has confused both myself, as well as at least [one other user](https://discourse.nixos.org/t/how-to-run-latest-anki-download/25406/5?u=cdepillabout).

With this PR, you can see that `anki-bin` now has a version set correctly:

```
$ nix-build -A anki-bin
/nix/store/a58frbn0p8yq6laa4rh8z91ajq4bpzsz-anki-bin-2.1.60
```

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/anki/bin.nix [hunk 1]
```diff
@@ -49,28 +49,39 @@ let
...
+
+    inherit meta passthru;
+  });
+
+  fhsUserEnvAnkiWithVersion = fhsUserEnvAnki.overrideAttrs (oldAttrs: {
+    # buildFHSUserEnv doesn't have an easy way to set the version of the
+    # resulting derivation, so we manually override it here.  This makes
+    # it clear to end users the version of anki-bin.  Without this, users
+    # might assume anki-bin is an old version of Anki.
+    name = "${pname}-${version}";
```
Comment (cdepillabout) [inline pkgs/games/anki/bin.nix:79]: This changes the resulting name of the derivation from `anki` to `anki-bin-2.1.60`.

I don't have a strong opinion whether the derivation should be named `anki-2.1.60` or `anki-bin-2.1.60`.  I'd be fine with either.  (But I do feel strongly that the version should be included.)

**Comment by cdepillabout** [comment]
@ofborg build anki-bin

**Comment by mahmoudk1000** [comment]
I don't believe, that is necessary as the version can be seen from with in Anki setting itself.
In addition, the commit is not clean also; the indentation is heavily edited.
But generally I don't mind the idea!

**Comment by Atemu** [comment]
I'm also lukewarm on this. This should be fixed in buildFHSUserEnv, don't you think?

**Comment by cdepillabout** [comment]
> I don't believe, that is necessary as the version can be seen from with in Anki setting itself.

It is important that the version is visible from the derivation, since it is used in things like `nix search`:

```console
$ nix search nixpkgs anki
* legacyPackages.x86_64-linux.anki (2.1.15)
  Spaced repetition flashcard program

* legacyPackages.x86_64-linux.anki-bin
  Spaced repetition flashcard program
```

You can see that when searching against current `master`, the version for `anki-bin` is not visible.

When when searching against this PR, the version is now visible:

```console
$ nix search ./. anki
* legacyPackages.x86_64-linux.anki (2.1.15)
  Spaced repetition flashcard program

* legacyPackages.x86_64-linux.anki-bin (2.1.60)
  Spaced repetition flashcard program
```

**Comment by cdepillabout** [comment]
> This should be fixed in buildFHSUserEnv, don't you think?

That makes sense.  The reason I didn't do that for two reasons:

1. I want to backport this to 22.11, but I don't particularly want to deal with backporting a change to `buildFHSUserEnv`.
2. I thought there was a possibility that adding a version to `buildFHSUserEnv` could lead to some annoying bike-shedding back and forth.

How about this: 

1. We merge this PR in (as-is?).
2. I'll backport this PR to 22.11.
3. I'll open a follow-up PR to `buildFHSUserEnv` adding a `version` argument.
4. When (if?) that PR gets merged in, I'll send a follow-up PR to `anki-bin` switching it over to using the new `version` on `buildFHSUserEnv`.

How does that sound?

**Comment by Atemu** [comment]
That sounds good to me. An alternative would be to bring this PR only to 22.11 and have the proper fix in master.

**Comment by cdepillabout** [comment]
> An alternative would be to bring this PR only to 22.11 and have the proper fix in master.

I'd prefer to get this PR into `master`, even if it is arguably not the best fix, and then improve it in a follow-up PR.

But if you absolutely want to avoid merging this PR, then I can understand.

**Comment by cdepillabout** [comment]
@Atemu 

>  But it looks like you just don't have merge rights

Let's try to fix that :-)

https://github.com/NixOS/nixpkgs/issues/50105#issuecomment-1445086175

**Pull request merged**